### PR TITLE
Implement project commands, improve app responsiveness, prevent ConflictError raised while changing UI values quickly

### PR DESCRIPTION
**Pull request by ansRMorin**
## What type of PR is this? (check all applicable)

- [x] Refactoring
- [x] User story with back-end changes
- [x] User story with front-end changes
- [ ] Bug Fix
- [ ] Documentation Update

## Description
The purpose of this PR is to implement new feature of disabling the project command buttons while command execution method is in progress. This PR includes:
- Refactoring of command execution methods to a single transaction method to handle all action requests on the oSL session.
- Created a ButtonGroup and ButtonWithTooltip class to simplify project command group creation for the UI
- Removed the ActorCommandAIO class and refactored the ProjectCommandsAIO class to handle all button interactions for the root and actor commands.
- Updated version of glow to 0.4.dev4 --> significant improvement in the App's responsiveness
- Updated version of portal to 0.4.3
- Handles ConflictError thrown when changing UI values quickly

## Related Tickets

- Closes https://github.com/Solution-Applications/solution-enablement-product-management/issues/581
- Closes https://github.com/Solution-Applications/solution-enablement-product-management/issues/548
- Fixes problem seen in: https://github.com/Solution-Applications/solution-enablement-product-management/issues/546

## QA Instructions,

1. Checkout branch associated to this PR
2. In a .venv run: flit install to gather latest changes made to the osl-solution template
3. Create a new osl-solution using overview_project.owa: ansys-templates new solution -f owa (enter all required inputs)
4. In your terminal, set your current directory to the directory of the new solution.
5.  Setup the python virtual environment by running: python setup_environment.py -d run
6.  Run the solution and create a new project
7. Start the analysis
8. Navigate to the root component of the monitoring pages. Click on the project commands and check the all buttons are disabled while the method is running. Check the an alert is displayed once method execution is complete.  
9. Repeat step 8 using the actor commands.

### CODE REVIEW

**Review by iazehaf — CHANGES_REQUESTED** [review]
@ansRMorin the button_group.py is not added to the desired structure of the cookicutter template. It's missing in the desired project structure.

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/components/commands.py [hunk 1]
```diff
@@ -1,46 +1,24 @@
...
+
 import uuid
 
 from ansys.saf.glow.client.dashclient import DashClient
-from ansys.solutions.{{ cookiecutter.__solution_name_slug }}.solution.problem_setup_step import ProblemSetupStep
-from ansys.solutions.{{ cookiecutter.__solution_name_slug }}.solution.definition import {{ cookiecutter.__solution_definition_name }}
+from ansys.saf.glow.solution import MethodStatus
 
+from ansys.solutions.oscillator_progress.solution.definition import Oscillator_ProgressSolution
+from ansys.solutions.oscillator_progress.ui.components.button_group import ButtonGroup
```
[the remark this answers is not in the document]
Comment (ansRMorin) [reply src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/components/commands.py:14]: good catch, done!

**Comment by ansRMorin** [comment]
I had to update version of GLOW to 0.4.dev4 to get the MethodStatus for a normal transaction method. However, when I run the solution I get this error:
image
This is because GLOW upgrade to pydantic v2 and portal has not yet. To test, you can run the solution with `saf run --project <project_name>`

**Review by iazehaf — CHANGES_REQUESTED** [review]
@ansRMorin `problem_setup_step` needs to be replaced by `self` [here](https://github.com/ansys/ansys-templates/blob/d875bc29401229365878cef8d0da090da1bab4f5/src/ansys/templates/python/osl_solution/%7B%7Bcookiecutter.__project_name_slug%7D%7D/src/ansys/solutions/%7B%7Bcookiecutter.__solution_name_slug%7D%7D/solution/problem_setup_step.py#L427C36-L427C36). 

Commands locking works fine but I see no alert message when the command is running. The purpose of the alert message is to indicate to the user which command is running on which node. 

I recommend we rename the method `run_selected_project_command` into `control_node_state`.

I think Portal 0.4.3 is compatible with GLOW 0.4.dev4.

**Comment by ansRMorin** [comment]
> @ansRMorin `problem_setup_step` needs to be replaced by `self` [here](https://github.com/ansys/ansys-templates/blob/d875bc29401229365878cef8d0da090da1bab4f5/src/ansys/templates/python/osl_solution/%7B%7Bcookiecutter.__project_name_slug%7D%7D/src/ansys/solutions/%7B%7Bcookiecutter.__solution_name_slug%7D%7D/solution/problem_setup_step.py#L427C36-L427C36).
>_done!_
> 
> Commands locking works fine but I see no alert message when the command is running. The purpose of the alert message is to indicate to the user which command is running on which node.
>_This can maybe be done at a later time in another PR. The fix is not so simple because you can't reliably update the same property of a component with two transaction methods. I would have to use a similar approach that I used to update the state of the buttons. One transaction updates the entire button group the second transaction method updates the state of each inidividual button (seems like I'm cheating the system here)_

> I recommend we rename the method `run_selected_project_command` into `control_node_state`.
_done!_

> I think Portal 0.4.3 is compatible with GLOW 0.4.dev4.
_I updated the portal to 0.4.3, but now I'm seeing this issue when I run the solution:_
image

**Comment by iazehaf** [comment]
> > @ansRMorin `problem_setup_step` needs to be replaced by `self` [here](https://github.com/ansys/ansys-templates/blob/d875bc29401229365878cef8d0da090da1bab4f5/src/ansys/templates/python/osl_solution/%7B%7Bcookiecutter.__project_name_slug%7D%7D/src/ansys/solutions/%7B%7Bcookiecutter.__solution_name_slug%7D%7D/solution/problem_setup_step.py#L427C36-L427C36).
> > _done!_
> > Commands locking works fine but I see no alert message when the command is running. The purpose of the alert message is to indicate to the user which command is running on which node.
> > _This can maybe be done at a later time in another PR. The fix is not so simple because you can't reliably update the same property of a component with two transaction methods. I would have to use a similar approach that I used to update the state of the buttons. One transaction updates the entire button group the second transaction method updates the state of each inidividual button (seems like I'm cheating the system here)_
> 
> > I recommend we rename the method `run_selected_project_command` into `control_node_state`.
> > _done!_
> 
> > I think Portal 0.4.3 is compatible with GLOW 0.4.dev4.
> > _I updated the portal to 0.4.3, but now I'm seeing this issue when I run the solution:_
> > image

@ansRMorin this is due to a bug on Portal 0.4.3 wheelhouse. Switching yo 0.5.dev0 resolves the issue. We are good to merge this PR. Nice work here!

**Review by iazehaf — APPROVED** [review]

**Pull request merged**